### DISCUSSION TITLE
Spike prime hub multitasking issue, and the robot not being consistent in the acceleration and the moving forward.

### DISCUSSION THREAD

**Discussion by Meki-Zeyad** [pybricks/support#1937] [Q&A] [answered]
so i have been using pybricks for a while now didnt see in issues/troubles using it untill it came to multitasking function, i wanted the robot to do more than one thing at the same time i will just drop the code here so you could understand 

```python
# first try
async def first_move():

    drive_base.settings(1000, 100, 150, 50) # setting the speed and acceleration
    multitask(drive_base.straight(-200, Stop.HOLD), motor_degrees(f_motor, 500, -500))
async def second_move():
    
    drive_base.straight(200, Stop.HOLD)

run_task(first_move())
run_task(second_move())

# second try
async def first_move():

    drive_base.settings(1000, 100, 150, 50) # setting the speed and acceleration
    multitask(drive_base.straight(-200, Stop.HOLD), motor_degrees(f_motor, 500, -500))

run_task(first_move())
drive_base.straight(200, Stop.HOLD)

# third try 

async def first_move():

    drive_base.settings(1000, 100, 150, 50) # setting the speed and acceleration
    multitask(drive_base.straight(-200, Stop.HOLD), motor_degrees(f_motor, 500, -500))
    wait(100)
    drive_base.straight(200, Stop.HOLD)

run_task(first_move())
```
i tryed alot more but i just gave some examples, but the thing am trying to do here is after the multitask() function is done i move with anyother movment but it just do the first_move() function and then gives me the error even if i do 2 different functions and use run_task()

```
the error: (Traceback (most recent call last):
  File "solo.py", line 163, in <module>
  File "solo.py", line 160, in main
TypeError: 'NoneType' object isn't iterable
Traceback (most recent call last):
  File "solo.py", line 163, in <module>
  File "solo.py", line 160, in main
TypeError: 'NoneType' object isn't iterable
Traceback (most recent call last):
  File "solo.py", line 165, in <module>
  File "solo.py", line 160, in first_move
TypeError: 'NoneType' object isn't iterable)
```
this is the 3 of them

i will really appreciate the help, Thank you 

, and the second part about "and the robot not being consistent in the acceleration and the moving forward"
i didnt visually see it but my coach said that happend alot

**Comment by laurensvalk** [comment]
Hi @Meki-Zeyad, a great way to see how to use multi-tasking is to try it with block coding first. Then you can see the generated code to see how it looks in Python. See also [this Python example](https://docs.pybricks.com/en/latest/tools/index.html#pybricks.tools.run_task). I would recommend starting with a working example and modify it as you need for your application.

**Reply by Meki-Zeyad** [reply]
ok yes i can see, but i think didnt explain it right. 

so for example here:
async def main():
    await multitask(drive_base.straight(400, Stop.HOLD), motor_degrees(f_motor, 400, 1000))
    await drive_base.straight(400, Stop.HOLD)



run_task(main())

it works untill it comes to the "await drive_base.straight(400, Stop.HOLD)"

i get this error when i ever i try to run it or when it reaches the code line 
the error:

TypeError: 'NoneType' object isn't iterable
Traceback (most recent call last):
  File "solo.py", line 198, in <module>
  File "solo.py", line 193, in main
TypeError: 'NoneType' object isn't iterable

**Reply by laurensvalk** [reply, marked as answer]
Right, so have a careful look at the required syntax for multitasking code.

If you make your own functions, like `motor_degrees`, remember that they should use `async def` just like all the other functions.

**Reply by Meki-Zeyad** [reply]
no no my proplem isnt with the motor_dgrees that works just fine, but what after that it dosnet work.

edit:
(Oh oh am sorry it worked thank you, and am sorry for wasting your time)
